src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1266,6 +1266,27 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             return LowerHWIntrinsicCmpOp(node, GT_NE);
         }
 
+        case NI_Sve_TestAnyTrue:
```

Comment:
What is happening here? Is it ensuring the bool return is set set?

Reply:
Because `TestAnyTrue` is tied to the instruction `ptest`, the instruction itself doesn't have a destination register; it only sets the conditional flags.

This lowering transformation effectively handles the conditional flags and returns the appropriate 'bool' value we expect. Changing `TestAnyTrue`'s `gtType` to `TYP_VOID` ensures we won't allocate a destination register for that particular node.

Reply:
> Changing TestAnyTrue's gtType to TYP_VOID ensures we won't allocate a destination register for that particular node.

I think I get this part. What I am trying to understand is how we make sure that the underlying operation is doing what it is supposed to do:

- [TestAnyTrue](https://developer.arm.com/architectures/instruction-sets/intrinsics/svptest_any): Return true if at least one element is active and if at least one active element of op is true.
- [TestFirstTrue](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svptest_first):  Return true if at least one element is active and if the first active element of op is true.
- [TestLastTrue](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svptest_last): Return true if at least one element is active and if the last active element of op is true.

Can you share the disassembly of each of those?